base/reduce.jl

```diff
@@ -1200,10 +1200,12 @@ count(itr; init=0) = count(identity, itr; init)
 
 count(f, itr; init=0) = _simple_count(f, itr, init)
 
-function _simple_count(pred, itr, init::T) where {T}
+_simple_count(pred, itr, init) = _simple_count_helper(Generator(pred, itr), init)
+
+function _simple_count_helper(g, init::T) where {T}
     n::T = init
-    for x in itr
-        n += pred(x)::Bool
+    for x in g
+        n += x::Bool
```

Comment:
Is this just `n += true` now?

Reply:
no `pred` can return `true` or `false`, e.g.:
```
julia> collect(Base.Generator(ismissing, [1,missing,2]))
3-element Vector{Bool}:
 0
 1
 0
```
and we still need `::Bool` assertion

Reply:
Oh yeah, somehow I thought the first argument was a filter with this syntax.

Reply:
That is why I was surprised that the compiler does not vectorize it (the only thing I do is introduce `Generator` wrapper and add a function barrier - but both are needed to force vectorization)